BenchPress/Generators.Tests/TestMetadataTests.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace Generators.Tests
+{
+    [TestClass]
+    public class TestMetadataTests
+  {
+      [TestMethod]
+      public void Should_Throw_If_Unknown_ResourceType()
+      {
+          // arrange
+          var resourceType = "Unknown";
+          var resourceName = "Name";
+          var extraProperties = new Dictionary<string, string>();
+
+          // act and assert
+          Action action = () => { new TestMetadata(resourceType, resourceName, extraProperties); };
+          Assert.ThrowsException<UnknownResourceTypeException>(action);
```

Comment:
A couple of notes for these two tests.

The first one is that this test doesn't test `TestMetadata.cs`, it tests `ResourceType,cs` indirectly through the `TestMetadata` constructor. This test should be refactored for use in a test for `ResourceType`.

The other is that constructors, typically, don't fit well into unit tests unless there is some complex logic in the constructor. In which case, the constructor needs to be reevaluated. With that said, the below test should probably be removed, because combined with the above comment it is only testing the ability to assign variables.

Reply:
true, I refactored to remove `TestMetadataTests.cs` and replaced it with `ResourceTypeTests.cs`